projects/material-dayjs-adapter/src/lib/adapter/dayjs-date-adapter.ts

```diff
@@ -135,16 +143,18 @@ export class DayjsDateAdapter extends DateAdapter<Dayjs> {
     return this.dayJs();
   }
 
-  // TODO: Handle locales here.
   parse(value: any, parseFormat: string): Dayjs | null {
-    return value ? this.dayJs(value, parseFormat) : null;
+    if (value && typeof value == 'string') {
+      return this.dayJs(value, dayjs().localeData().longDateFormat(parseFormat), this.locale);
+    }
+    return value ? this.dayJs(value).locale(this.locale) : null;
   }
 
   format(date: Dayjs, displayFormat: string): string {
     if (!this.isValid(date)) {
-      throw Error('MomentDateAdapter: Cannot format invalid date.');
+      throw Error('DayjsDateAdapter: Cannot format invalid date.');
     }
-    return date.format(displayFormat);
+    return date.locale(this.locale).format(displayFormat);
```

Comment:
I wonder if this is required here. Maybe this works better if we keep optional locales in mind:
```suggestion
    return this.locale ? date.locale(this.locale).format(displayFormat) : date.format(displayFormat);
```

Reply:
The locale is set by default to "en_US" by Angular Material. 
So checking locale for falsiness will not be necessary.